From 615b0784516933106a8446272bc3c043b0a0d50a Mon Sep 17 00:00:00 2001
From: Ken Gaillot <kgaillot@redhat.com>
Date: Tue, 26 Jan 2016 14:04:30 -0600
Subject: [PATCH] Fix: crmd: set remote flag when gracefully shutting down
 remote nodes

---
 crmd/lrm_state.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/crmd/lrm_state.c b/crmd/lrm_state.c
index 62e1c76..7ea4e8a 100644
--- a/crmd/lrm_state.c
+++ b/crmd/lrm_state.c
@@ -488,7 +488,7 @@ remote_proxy_cb(lrmd_t *lrmd, void *userdata, xmlNode *msg)
         crm_notice("Graceful proxy shutdown of %s", lrm_state->node_name);
 
         now_s = crm_itoa(now);
-        update_attrd(lrm_state->node_name, XML_CIB_ATTR_SHUTDOWN, now_s, NULL, FALSE);
+        update_attrd(lrm_state->node_name, XML_CIB_ATTR_SHUTDOWN, now_s, NULL, TRUE);
         free(now_s);
 
         remote_proxy_ack_shutdown(lrmd);
-- 
1.8.3.1

