From d21fc64b7ff6333a4bddbc6dca401dacc1541d05 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Jan=20Pokorn=C3=BD?= <jpokorny@redhat.com>
Date: Tue, 9 Feb 2016 11:11:38 +0100
Subject: [PATCH] Build: make sure PACKAGE_URL is indeed defined

---
 configure.ac | 7 ++++++-
 version.m4   | 1 +
 2 files changed, 7 insertions(+), 1 deletion(-)

diff --git a/configure.ac b/configure.ac
index cc9dfe7..4c0b7a1 100644
--- a/configure.ac
+++ b/configure.ac
@@ -20,7 +20,12 @@ dnl     checks for library functions
 dnl     checks for system services
 
 m4_include([version.m4])
-AC_INIT([pacemaker], VERSION_NUMBER, pacemaker@oss.clusterlabs.org,pacemaker,http://clusterlabs.org)
+AC_INIT([pacemaker], VERSION_NUMBER, [pacemaker@oss.clusterlabs.org],
+        [pacemaker], PCMK_URL)
+dnl Workaround autoconf < 2.64
+if test x"${PACKAGE_URL}" = x""; then
+	AC_SUBST([PACKAGE_URL], PCMK_URL)
+fi
 
 PCMK_FEATURES=""
 HB_PKG=heartbeat
diff --git a/version.m4 b/version.m4
index 70596af..d5aaeb4 100644
--- a/version.m4
+++ b/version.m4
@@ -1 +1,2 @@
 m4_define([VERSION_NUMBER], [1.1.14])
+m4_define([PCMK_URL], [http://clusterlabs.org])
-- 
1.8.3.1

