From 5743dda7ea27f3e4626e6b676fd7d68888adb0d2 Mon Sep 17 00:00:00 2001
From: Ken Gaillot <kgaillot@redhat.com>
Date: Mon, 21 Mar 2016 17:44:44 -0500
Subject: [PATCH] Feature: tools: crm_mon now sanitizes logfiles

---
 tools/report.collector | 15 +++++++++------
 1 file changed, 9 insertions(+), 6 deletions(-)

diff --git a/tools/report.collector b/tools/report.collector
index 0cf4a1e..ecd1546 100644
--- a/tools/report.collector
+++ b/tools/report.collector
@@ -728,7 +728,8 @@ for p in $LOG_PATTERNS; do
 done > $pattfile
 
 for l in $logfiles $EXTRA_LOGS; do
-    b=`basename $l`
+    b="$(basename $l).extract.txt"
+
     if [ ! -f "$l" ]; then
 	# Not a file
 	continue
@@ -739,7 +740,10 @@ for l in $logfiles $EXTRA_LOGS; do
     fi
+
     dumplogset "$l" $LOG_START $LOG_END > "$b"
+    sanitize "$b"
+
     echo "Log patterns $REPORT_TARGET:"  > $ANALYSIS_F
-    cat $b | grep -f $pattfile >> $ANALYSIS_F
+    grep -f "$pattfile" "$b" >> $ANALYSIS_F
 done
 
 which journalctl > /dev/null 2>&1
@@ -769,12 +772,12 @@ done
 
 # Parse for events
 for l in $logfiles $EXTRA_LOGS; do
-    node_events `basename $l` > $EVENTS_F
+    b="$(basename $l).extract.txt"
+    node_events "$b" > $EVENTS_F
 
     # Link the first logfile to a standard name if it doesn't yet exist
-    f=`basename $l`
-    if [ -e $f -a ! -e $HALOG_F ]; then
-	ln -s $f $HALOG_F
+    if [ -e "$b" -a ! -e "$HALOG_F" ]; then
+	ln -s "$b" "$HALOG_F"
     fi
 done
 
-- 
1.8.3.1

