From b47749de4916c6090d0e1593139553a84be97db1 Mon Sep 17 00:00:00 2001
From: Ken Gaillot <kgaillot@redhat.com>
Date: Fri, 30 Nov 2018 16:39:53 -0600
Subject: [PATCH] Test: cts-cli: update regression test for all_stopped removal

---
 tools/regression.tools.exp | 2 --
 1 file changed, 2 deletions(-)

diff --git a/tools/regression.tools.exp b/tools/regression.tools.exp
index ce9f352..f5a2a42 100644
--- a/tools/regression.tools.exp
+++ b/tools/regression.tools.exp
@@ -1805,7 +1805,6 @@ Executing cluster transition:
  * Resource action: Fence           stop on node1
  * Resource action: Fence           monitor on node3
  * Resource action: Fence           monitor on node2
- * Pseudo action:   all_stopped
  * Resource action: Fence           start on node2
 
 Revised cluster status:
@@ -2074,7 +2073,6 @@ Transition Summary:
 
 Executing cluster transition:
  * Resource action: dummy           stop on node1
- * Pseudo action:   all_stopped
  * Resource action: dummy           start on node3
 
 Revised cluster status:
-- 
1.8.3.1

