From 31db95be6e32408951e6b9c26ead1d0247ba5413 Mon Sep 17 00:00:00 2001
From: Ken Gaillot <kgaillot@redhat.com>
Date: Wed, 6 Jul 2016 16:06:58 -0500
Subject: [PATCH] Fix: libcrmcommon: remove extraneous format specifier from
 log message

Bug introduced in 1.1.12 via c20c553
---
 lib/common/xml.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/lib/common/xml.c b/lib/common/xml.c
index a20e73e..3e3186e 100644
--- a/lib/common/xml.c
+++ b/lib/common/xml.c
@@ -4329,7 +4329,7 @@ __xml_diff_object(xmlNode * old, xmlNode * new)
             xml_private_t *p = new_child->_private;
 
             if(p_old != p_new) {
-                crm_info("%s.%s moved from %d to %d - %d",
+                crm_info("%s.%s moved from %d to %d",
                          new_child->name, ID(new_child), p_old, p_new);
                 __xml_node_dirty(new);
                 p->flags |= xpf_moved;
-- 
1.8.3.1

